### ISSUE TITLE
Synchronize by Git file list

### ISSUE THREAD

**Issue by Vakhrushev** [fioan89/sourcesync#33]
- Have some amount commits.
* 066e8ab (HEAD -> test, origin/test, origin/dev, dev) #29938
* 30cefb5 #29937
* 44dd3c1 #29933
* db61f14 #22223
* c1f3d14 #29844
* 08d2a2d #29844

First case.
Need sync all changed files between  c1f3d14 and HEAD

Second case
Need sync all files between 08d2a2d and 44dd3c1

is it posible ?

**Comment by fioan89** [comment]
Yes it is possible. But it will take quite a long time to implement this. In the most optimistic case I would say somewhere around two months. Reason being that I have to do a spike for the VCS API that IDEA exposes. This is not publicly documented by Jetbrains - or at least I failed to find good, explicit documentation on this. So I'll have to crunch through IntelliJ source code which is fine by me, but it will require a good chunk of time. After the spike is done and I understand how the VCS mechanism works, I will dive into implementing the feature. This will require a major rewrite of the UI - which was not designed for this kind of jobs in mind.

Case in point is that I prioritize work on the time spent and effort that needs to be done to implement the issue. Those that require less work and time to fix will be the first that I implement. This needs quite a lot of time/effort so it will probably go to the bottom of my stack :(

I know it's not a good way of doing things but this is the way I manage to work on the plugin in the little free time that I have.
